### PR TITLE
Services/biblio ref/add halucinated

### PR DESCRIPTION
**Pull request by leogail**
- added an halucinated status
- commented code

:warning: :warning: :warning:  have to merge #54 before

### CODE REVIEW

File: services/biblio-ref/v1/bibref/bibref_functions.py [hunk 1]
```diff
@@ -0,0 +1,199 @@
+import re
+import unicodedata
+import re
+from requests_ratelimiter import LimiterSession
+import unicodedata
+from thefuzz import fuzz
+
+
+mail_adress = "leo.gaillard@cnrs.fr"
```
Comment (leogail) [inline services/biblio-ref/v1/bibref/bibref_functions.py:9]: **praise (non-blocking):** 
```suggestion
mail_address = "leo.gaillard@cnrs.fr"
```

File: services/biblio-ref/v1/bibref/bibref_functions.py [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+def verify_doi(doi, mail=mail_adress):
+    """
+    Verify a Digital Object Identifier (DOI) by making a GET request to the Crossref API. 
+
+    Args:
+        doi (str): The DOI to be verified.
+        mail (str): The email address to be included in the API request. Defaults to the value of mail_address.
+
+    Returns:
+        int: The HTTP status code of the API response, or 503 if an exception occurs.
```
Comment (leogail) [inline services/biblio-ref/v1/bibref/bibref_functions.py:82]: **praise:** 
```suggestion
        (int,dict): 
        - the first element is the HTTP status code of the API response, or 503 if an exception occurs.
        - the dict is informations found with request. if not found, every fields of the dictionay are empty
```

File: services/biblio-ref/v1/bibref/bibref_functions.py [hunk 3]
```diff
@@ -0,0 +1,199 @@
...
+    """
+    check with crossref api if a biblio ref is correct.
+    Objective : In this function, we check if the biblio ref exist if no doi is found : we use as critera the compare_pubinfo_refbiblio function
+    Args : 
+        ref_biblio :a biblio ref
+        mail : a mail adress
+    
+    Returns : 
+        a confidence score about the existence + doi of the biblio ref
+    """
```
Comment (leogail) [inline services/biblio-ref/v1/bibref/bibref_functions.py:179]: **praise (non-blocking):** 
```suggestion
def verify_biblio(ref_biblio, mail=mail_adress):

    """
    check with crossref api if a biblio ref is correct.
    Objective : In this function, we check if the biblio ref exist if no doi is found : we use as critera the compare_pubinfo_refbiblio function
    
    Args : 
        ref_biblio :a biblio ref
        mail : a mail adress
    
    Returns : 
        a confidence score about the existence + doi of the biblio ref
    """
    
```

[the remark this answers is not in the document]
Comment (leogail) [reply services/biblio-ref/v1/bibref/bibref_functions.py]: done on all

File: services/biblio-ref/v1/bibref/bibref_functions.py [hunk 4]
```diff
@@ -0,0 +1,199 @@
...
+            others_biblio_info["first_author_name"] = ""
+        try:
+            others_biblio_info["first_author_given"] = message['author'][0]['given']
+        except:
+            others_biblio_info["first_author_given"] = ""
+            
+        return (response.status_code , others_biblio_info)
+
+    except:
+        return (503,{'title': "", 'first_author_given': "", 'first_author_name': "", 'doi': ""}) # if there is an unexpected error from crossref
```
[the remark this answers is not in the document]
Comment (leogail) [reply services/biblio-ref/v1/bibref/bibref_functions.py:113]: done on all

**Review by leogail — COMMENTED** [review]
commit suggestions of @Luc-Ank and @parmentf 

Except isinstance function which is a python function, not mine

**Review by parmentf — APPROVED** [review]

**Pull request merged**